### PR TITLE
log: hide password in log

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
- replace password with '******' in log

NOTE: support dumpling later

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Comment by lance6716** [comment]
busy CI 😂 
```
runtime: failed to create new OS thread (have 21 already; errno=12)
fatal error: newosproc
```

File: tests/all_mode/run.sh [hunk 1]
```diff
@@ -151,6 +151,15 @@ function run() {
     ls $WORK_DIR/worker1/dumped_data.test && exit 1 || echo "worker1 auto removed dump files"
     ls $WORK_DIR/worker2/dumped_data.test && exit 1 || echo "worker2 auto removed dump files"
 
+    echo "check no password in log"
+    check_log_not_contains $WORK_DIR/master/log/dm-master.log "/Q7B9DizNLLTTfiZHv9WoEAKamfpIUs="
+    check_log_not_contains $WORK_DIR/worker1/log/dm-worker.log "/Q7B9DizNLLTTfiZHv9WoEAKamfpIUs="
+    check_log_not_contains $WORK_DIR/worker2/log/dm-worker.log "/Q7B9DizNLLTTfiZHv9WoEAKamfpIUs="
+    check_log_not_contains $WORK_DIR/master/log/dm-master.log "123456"
+   # uncomment after hide password in dumpling
```
Comment (csuzhangxc) [inline tests/all_mode/run.sh:159]: open an issue in dumpling repo?

**Review by csuzhangxc — COMMENTED** [review]
LGTM

File: pkg/utils/util.go [hunk 2]
```diff
@@ -87,11 +87,15 @@ var (
 		"^SET\\s+PASSWORD",
 	}
 	builtInSkipDDLPatterns *regexp.Regexp
+
+	passwordPatterns = `(password: (\\")?)(.*?)((\\")?\\n)`
```
Comment (lance6716) [inline pkg/utils/util.go:91]: Why so many group 😂

**Review by lance6716 — COMMENTED** [review]
rest LGTM

**Comment by lance6716** [comment]
LGTM

File: dm/master/server.go [hunk 3]
```diff
@@ -425,7 +425,7 @@ func (s *Server) StartTask(ctx context.Context, req *pb.StartTaskRequest) (*pb.S
 		resp.Msg = err.Error()
 		return resp, nil
 	}
-	log.L().Info("", zap.String("task name", cfg.Name), zap.Stringer("task", cfg), zap.String("request", "StartTask"))
+	log.L().Info("", zap.String("task name", cfg.Name), zap.String("task", cfg.JSON()), zap.String("request", "StartTask"))
```
[the remark this answers is not in the document]
Comment (GMHDBJD) [reply dm/master/server.go:428]: We pass task config as string many places, I am afraid to change all of them.😪

**Pull request merged**